sound/soc/sof/intel/hda-ctrl.c

```diff
@@ -139,20 +139,19 @@ void hda_dsp_ctrl_misc_clock_gating(struct snd_sof_dev *sdev, bool enable)
  */
 int hda_dsp_ctrl_clock_power_gating(struct snd_sof_dev *sdev, bool enable)
 {
-#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
-	struct hdac_bus *bus = sof_to_bus(sdev);
-#endif
 	u32 val;
 
 	/* enable/disable audio dsp clock gating */
 	val = enable ? PCI_CGCTL_ADSPDCGE : 0;
 	snd_sof_pci_update_bits(sdev, PCI_CGCTL, PCI_CGCTL_ADSPDCGE, val);
 
-#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
-	/* enable/disable L1 support */
-	val = enable ? SOF_HDA_VS_EM2_L1SEN : 0;
-	snd_hdac_chip_updatel(bus, VS_EM2, SOF_HDA_VS_EM2_L1SEN, val);
-#endif
+	/*
+	 * Disable L1 support. Workaround to address a known issue with
+	 * host DMA that results in xruns during pause/release scenario.
+	 */
+	if (!enable)
```

Comment:
I would remove this "if (!enable)" condition to make the logic more clear -- disable L1 support always.

Reply:
@keyonjie fixed both comments. I've moved the L1 disabling step to hda_dsp_pre_fw_run() so it getts called only once.